flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -480,21 +480,28 @@ class OmpAttributeVisitor : DirectiveAttributeVisitor<llvm::omp::Directive> {
   bool Pre(const parser::OmpClause::Reduction &x) {
     const parser::OmpReductionOperator &opr{
         std::get<parser::OmpReductionOperator>(x.v.t)};
+    auto createDummyProcSymbol = [&](const parser::Name *name) {
+      // If name resolution failed, create a dummy symbol
+      const auto namePair{
+          currScope().try_emplace(name->source, Attrs{}, ProcEntityDetails{})};
+      auto &newSymbol{*namePair.first->second};
+      name->symbol = &newSymbol;
```

Comment:
nit: either don't create a new variable here, or just leave it as a pointer rather than dereferencing and then referencing it?
E.g. 
```
auto *newSymbol{namePair.first->second};
name->symbol = newSymbol;
```
I was a little confused by it turning into a reference and then back into a pointer.

Reply:
The change here is only moving code. We can fix this in a follow-up patch if required.
Also, note that `namePair.first->second`  is of type `Fortran::common::Reference<Fortran::semantics::Symbol>` and hence the code you provided will not directly work.